src/OpenTelemetry/Trace/TracerProviderBuilderExtensions.cs

```diff
@@ -92,7 +92,7 @@ public static TracerProvider Build(this TracerProviderBuilder tracerProviderBuil
         /// <returns>Returns <see cref="TracerProviderBuilder"/> for chaining.</returns>
         internal static TracerProviderBuilder AddDiagnosticSourceInstrumentation<TInstrumentation>(
             this TracerProviderBuilder tracerProviderBuilder,
-            Func<ActivitySourceAdapter, TInstrumentation> instrumentationFactory)
+            Func<TInstrumentation> instrumentationFactory)
```

Comment:
`AddDiagnosticSourceInstrumentation` method can now be eliminated and simply use `AddInstrumentation`

Reply:
Removed `AddDiagnosticSourceInstrumentation` method